specs/core/0_beacon-chain.md

```diff
@@ -1088,28 +1083,50 @@ def get_domain(fork: Fork,
     ) * 2**32 + domain_type
 ```
 
+#### `get_bitfield_bit`
+
+```python
+def get_bitfield_bit(bitfield: bytes, i: int) -> int:
+    return (bitfield[i // 8] >> (7 - (i % 8))) % 2
+```
+
+#### `verify_bitfield`
+
+```python
+def verify_bitfield(bitfield: bytes, size: int) -> bool:
+    ```
+    Verify ``bitfield`` against the ``size``.
+    if len(bitfield) != (size + 7) // 8:
+        return False
+
+    for i in range(size + 1, size + size % 8 + 8):
+        if get_bitfield_bit(bitfield, i) != 0:
+            return False
+
+    return True
+```
+
 #### `verify_slashable_vote_data`
 
 ```python
-def verify_slashable_vote_data(state: BeaconState, vote_data: SlashableVoteData) -> bool:
-    if len(vote_data.custody_bit_0_indices) + len(vote_data.custody_bit_1_indices) > MAX_CASPER_VOTES:
+def verify_slashable_vote_data(state: BeaconState, slashable_vote_data: SlashableVoteData) -> bool:
+    if len(slashable_vote_data.custody_bit_1_indices) > 0:  # [TO BE REMOVED IN PHASE 1]
+        return False
+
+    if len(indices(slashable_vote_data)) > MAX_CASPER_VOTES:
         return False
 
-    return bls_verify_multiple(
+    return bls_verify(
         pubkeys=[
-            bls_aggregate_pubkeys([state.validator_registry[i].pubkey for i in vote_data.custody_bit_0_indices]),
-            bls_aggregate_pubkeys([state.validator_registry[i].pubkey for i in vote_data.custody_bit_1_indices]),
+            bls_aggregate_pubkeys([state.validator_registry[i].pubkey for i in slashable_vote_data.custody_bit_0_indices]),
+            bls_aggregate_pubkeys([state.validator_registry[i].pubkey for i in slashable_vote_data.custody_bit_1_indices]),
         ],
         messages=[
-            hash_tree_root(AttestationDataAndCustodyBit(vote_data.data, False)),
-            hash_tree_root(AttestationDataAndCustodyBit(vote_data.data, True)),
+            hash_tree_root(AttestationDataAndCustodyBit(slashable_vote_data.data, CUSTODY_BIT_ZERO)),
+            hash_tree_root(AttestationDataAndCustodyBit(slashable_vote_data.data, CUSTODY_BIT_ONE)),
```

Comment:
Is this really the cleanest way to express that zero and one are the custody bits? We could do it without adding new constants by doing `AttestationDataAndCustodyBit(data=slashable_vote_data.data, custody_bit=0)`; that way it requires the reader to hop around the spec less. I'm ultimately ok either way though.